packages/@aws-cdk/aws-stepfunctions-tasks/lib/athena/start-query-execution.ts

```diff
@@ -0,0 +1,259 @@
+            resource: 'catalog',
+          }),
+          cdk.Stack.of(this).formatArn({
+            service: 'glue',
+            resource: 'database',
+            resourceName: this.props.queryExecutionContext?.database ? this.props.queryExecutionContext?.database : 'default',
+          }),
+          cdk.Stack.of(this).formatArn({
+            service: 'glue',
+            resource: 'table',
+            resourceName: '*', // access needed to query all tables https://docs.aws.amazon.com/athena/latest/ug/security-iam-athena.html
+          }),
+          cdk.Stack.of(this).formatArn({
+            service: 'glue',
+            resource: 'userdefinedfunction',
+            resourceName: '*', // users need access to start user defined functions from step functions https://docs.aws.amazon.com/athena/latest/ug/security-iam-athena.html
+          }),
+        ],
+      }),
+    );
+
+    return policyStatements;
+  }
+
+  /**
+   * Provides the Athena start query execution service integration task configuration
+   */
+  /**
+   * @internal
+   */
+  protected _renderTask(): any {
+    return {
+      Resource: integrationResourceArn('athena', 'startQueryExecution', this.integrationPattern),
+      Parameters: sfn.FieldUtils.renderObject({
+        QueryString: this.props.queryString,
+        ClientRequestToken: this.props.clientRequestToken,
+        QueryExecutionContext: {
+          Catalog: this.props.queryExecutionContext?.catalog,
+          Database: this.props.queryExecutionContext?.database,
+        },
+        ResultConfiguration: {
+          EncryptionConfiguration: {
+            EncryptionOption: this.props.resultConfiguration?.encryptionConfiguration?.encryptionOption,
+            KmsKey: this.props.resultConfiguration?.encryptionConfiguration?.kmsKey,
+          },
+          OutputLocation: this.props.resultConfiguration?.outputLocation,
+        },
+        WorkGroup: this.props.workGroup,
+      }),
+    };
+  }
+}
+
+/**
+ * Location of query result along with S3 bucket configuration
+ *
+ * @see https://docs.aws.amazon.com/athena/latest/APIReference/API_ResultConfiguration.html
+ * @experimental
+ */
+export interface ResultConfiguration {
+
+  /**ResultConfigurationOptions
+   * S3 path of query results
+   *
+   * @default - No output location
+  */
+  readonly outputLocation?: string;
+
+  /**
+   * Encryption option used if enabled in S3
+   *
+   * @default - No encryption configuration
+   */
+  readonly encryptionConfiguration?: EncryptionConfiguration
+}
+
+/**
+ * Encryption Configuration of the S3 bucket
+ *
+ * @see https://docs.aws.amazon.com/athena/latest/APIReference/API_EncryptionConfiguration.html
+ * @experimental
+ */
+export interface EncryptionConfiguration {
+
+  /**
+   * Type of S3 server-side encryption enabled
+   *
+   * @default - 'SSE_S3'
+   */
+  readonly encryptionOption: string;
+
+  /**
+   * KMS key ARN or ID
+   *
+   * @default - No KMS key
+   */
+  readonly kmsKey?: kms.IKey;
```

Comment:
these are typically named `<resource>EncryptionKey` or simply `encryptionKey` - can you rename to whichever one fits better for the usage?